### PR TITLE
reworked player projectile

### PR DESCRIPTION
**Pull request by QbieShay**
With reduz's blessing, I reworked the projectile that the player fires in the tps demo. I am aware that it will need to be redone for 4.x, but that's a problem for future me (or future someone else) :D
Rework for monster laser will come somewhere later

I was suggested to add decals for the hit effect, but that requires some research, so it will probably come in a future PR :)

Here's a video of the new effect:
https://user-images.githubusercontent.com/7917475/103089520-7311a000-45ee-11eb-9b5c-cafb61e0d755.mp4

### CODE REVIEW

**Review by Calinou — APPROVED** [review]
Looks great!

**Comment by QbieShay** [comment]
> Looks great!

Thanks!

File: player/bullet/effect/BillboardPlane.glb.import [hunk 1]
```diff
@@ -0,0 +1,1063 @@
...
+
+[params]
+
+nodes/root_type="Spatial"
+nodes/root_name="Scene Root"
+nodes/root_scale=1.0
+nodes/custom_script=""
+nodes/storage=0
+materials/location=1
+materials/storage=1
```
Comment (aaronfranke) [inline player/bullet/effect/BillboardPlane.glb.import:20]:
```suggestion
materials/storage=2
```

Best to be explicit about storing the materials in a .tres file by changing `materials/storage` to `2`.

**Review by aaronfranke — COMMENTED** [review]
Looks awesome! I get some diffs when opening, can you delete your `.import/` folder and then re-open the demo, and also open and save some of the scenes again? (Ignore the door animation diffs, but I think anything else is worth committing).

File: player/bullet/effect/BillboardPlane.glb.import (hunk 1 above)
Comment (QbieShay) [reply player/bullet/effect/BillboardPlane.glb.import:20]: I don't think i store the materials externally for that file

**Comment by QbieShay** [comment]
@aaronfranke I think that I might have used 3.2.4, could that be the issue?

File: player/bullet/effect/BillboardPlane.glb.import (hunk 1 above)
Comment (aaronfranke) [reply player/bullet/effect/BillboardPlane.glb.import:20]: Then I suppose it should be set to 0 instead (built-in).

**Comment by aaronfranke** [comment]
@QbieShay Yes, these repos target the latest stable Godot version, which is currently Godot 3.2.3. Godot 3.2.4 is not out yet.

**Comment by QbieShay** [comment]
@aaronfranke I see. I'll fix it!

**Review by akien-mga — APPROVED** [review]

**Pull request merged**